File: TarjetaEstilosVueDasboard/src/models/Affiliates.ts
interface NetworkSelection {
  id: number,
  link: string
}


class Affiliates{
    title:string|null;
    image:string|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    id:number|null;
    networksIds:number[]|null=null 
    networks: NetworkSelection[]|null=null
    

    constructor(
        title:string|null,
        image:string|null,
        created_at:Date|null, updated_at:Date|null, deleted_at:Date|null, id:number|null=null,
        networksIds:number[]|null=null, 
        networks:NetworkSelection[]|null=null
    ){
        this.title = title;
        this.image = image;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.id = id;
        this.networksIds = networksIds;
        this.networks = networks;
    }
    
};
export default Affiliates;

File: src/main/java/pe/tarjetaestilos/models/Affiliates.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.ZonedDateTime;
import org.hibernate.annotations.Where;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "affiliates")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class Affiliates {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
            
    private String title; 
    
    private String image; 
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;

    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;
    
    
    @OneToMany(mappedBy = "affiliates")
    @JsonManagedReference
    @ToString.Exclude
    private List<AffiliatesNetworks> affiliatesNetworks = new ArrayList<>();
}
